src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -416,17 +426,18 @@ async Task OnIdle_QueueOnUIThread()
         }
 
         // Internal for testing
-#pragma warning disable VSTHRD100 // Avoid async void methods
-        internal async void OnResultsReady(object sender, BackgroundParserResultsReadyEventArgs args)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+        internal void OnResultsReady(object sender, BackgroundParserResultsReadyEventArgs args)
         {
             try
             {
-                UpdateParserState(args.CodeDocument, args.ChangeReference.Snapshot);
+                _joinableTaskContext.Factory.Run(async () =>
```

Comment:
This is different than async void like before. This will block the executing thread (the parsers thread) until it can swap fully to the UI thread

Reply:
swapped to `_ = OnResultsReadyAsync(...)`